lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -1364,8 +1364,14 @@ void request_evaluate(const llvm::json::Object &request) {
   std::string expression = GetString(arguments, "expression").str();
   llvm::StringRef context = GetString(arguments, "context");
 
-  if (context == "repl" && g_dap.DetectExpressionContext(frame, expression) ==
-                               ExpressionContext::Command) {
+  if (context == "repl" &&
+      (expression.empty() ?
+       g_dap.last_nonempty_var_expression.empty() :
+       g_dap.DetectExpressionContext(frame, expression) ==
```

Comment:
it's hard to read this. Please move this ternary check into another variable with a good name.

Reply:
.. but note that the work that DetectExpressionContext does (looking up local variables) is relatively expensive, so we probably don't want to do it when it's not needed (when `context!="repl"`).

Reply:
I've done the best compromise I can -- I've created a new variable, 'repeat_last_command' to use in the if-condition, and tried to set up the if-condition to avoid unnecessary calls to DetectExpressionContext.